src/gui/qgsfontbutton.cpp

```diff
@@ -320,6 +345,18 @@ void QgsFontButton::dragEnterEvent( QDragEnterEvent *e )
     e->acceptProposedAction();
     updatePreview( QColor(), nullptr, &font );
   }
+  else if ( mMode == ModeQFontColor && fontFromMimeData( e->mimeData(), font ) )
+  {
+    e->acceptProposedAction();
+    if ( colorFromMimeData( e->mimeData(), mimeColor, hasAlpha ) )
+    {
+      updatePreview( mimeColor, nullptr, &font );
+    }
+    else
+    {
+      updatePreview( QColor(), nullptr, &font );
+    }
```

Comment:
What about when just a color is dragged, not a font?

Reply:
Good point, I think it's handled now: https://github.com/qgis/QGIS/pull/47996/commits/5920bcd37ca3491500099db64d5bc3dc6c2c9d0b#diff-4bdad1c3f92a807479b512b850bde7e5299100c1ea092bb76de34cfd73caeb13R350